From e0e94ca613f5f38c2fcddb43aeb3274b9933b83d Mon Sep 17 00:00:00 2001
From: Tom St Denis <tom.stdenis@amd.com>
Date: Fri, 18 Dec 2015 08:35:48 -0500
Subject: [PATCH 0962/1050] drm/amd/powerplay:  Fix typo introduced by commit
 7036f23547f0b07ff33269fb0eec522fe0f8b5d1

The commit
   "drm/amd/powerplay: check whether need to enable thermal control."
lacks a compare for NULL in the constraint checking.

Signed-off-by: Tom St Denis <tom.stdenis@amd.com>
Reviewed-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c b/drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c
index 707664d..f0700d0 100644
--- a/drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c
+++ b/drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c
@@ -422,7 +422,7 @@ int pem_task_initialize_thermal_controller(struct pp_eventmgr *eventmgr, struct
 	range.max = TEMP_RANGE_MAX;
 	range.min = TEMP_RANGE_MIN;
 
-	if (eventmgr == NULL || eventmgr->platform_descriptor)
+	if (eventmgr == NULL || eventmgr->platform_descriptor == NULL)
 		return -EINVAL;
 
 	if (phm_cap_enabled(eventmgr->platform_descriptor->platformCaps, PHM_PlatformCaps_ThermalController))
-- 
1.9.1

